src/citrine/ara/columns.py

```diff
@@ -194,6 +209,111 @@ def __init__(self, *,
         self.data_source = data_source
 
 
+class FlatCompositionColumn(Serializable["FlatCompositionColumn"], Column):
+    """[ALPHA] Column that flattens the composition into a string of names and quantities.
+
+    Parameters
+    ----------
+    data_source: str
+        name of the variable to use when populating the column
+    sort_order: CompositionSortOrder
+        order with which to sort the components when generating the flat string
+
+    """
+
+    data_source = properties.String('data_source')
+    sort_order = properties.Enumeration(CompositionSortOrder, 'sort_order')
+    typ = properties.String('type', default="flat_composition_column", deserializable=False)
+
+    def _attrs(self) -> List[str]:
+        return ["data_source", "sort_order", "typ"]
+
+    def __init__(self, *,
+                 data_source: str,
+                 sort_order: CompositionSortOrder):
+        self.data_source = data_source
+        self.sort_order = sort_order
+
+
+class ComponentQuantityColumn(Serializable["ComponentQuantityColumn"], Column):
+    """[ALPHA] Column that extracts the quantity of a given component.
+
+    Parameters
+    ----------
+    data_source: str
+        name of the variable to use when populating the column
+    component_name: str
+        name of the component from which to extract the quantity
+
+    """
+
+    data_source = properties.String('data_source')
+    component_name = properties.String("component_name")
+    typ = properties.String('type', default="component_quantity_column", deserializable=False)
+
+    def _attrs(self) -> List[str]:
+        return ["data_source", "component_name", "typ"]
+
+    def __init__(self, *,
+                 data_source: str,
+                 component_name: str):
+        self.data_source = data_source
+        self.component_name = component_name
+
+
+class NthBiggestComponentNameColumn(Serializable["NthBiggestComponentNameColumn"], Column):
+    """[ALPHA] Name of the Nth biggest component.
+
+    Parameters
+    ----------
+    data_source: str
+        name of the variable to use when populating the column
+    n: int
+        index of the component name to extract, starting with 1 for the biggest
```

Comment:
What happens if `n` is larger than the number of components? For `NthBiggestComponentQuantityColumn` I would expect to get a value of 0, but I do not know what to expect here.

Reply:
Its empty.  I'll clarify in docs.